native/acter/src/api/common.rs

```diff
@@ -326,11 +331,46 @@ impl TryFrom<&AttachmentContent> for MsgContent {
     }
 }
 
+pub struct UrlPreview(pub(crate) RumaUrlPreview);
+
+impl UrlPreview {
+    pub fn from(prev: &RumaUrlPreview) -> Self {
+        Self(prev.clone())
+    }
+    pub fn new(prev: RumaUrlPreview) -> Self {
+        Self(prev)
+    }
+    pub fn url(&self) -> Option<String> {
+        self.0.url.clone()
+    }
+    pub fn title(&self) -> Option<String> {
+        self.0.title.clone()
+    }
+    pub fn description(&self) -> Option<String> {
+        self.0.description.clone()
+    }
+
+    pub fn has_image(&self) -> bool {
+        false // not yet supported
+              // !self.0.image.is_none()
+    }
+    pub fn image_source(&self) -> Option<MediaSource> {
+        None // not yet support
```

Comment:
should work but we are missing the right types from upstream. PR has already been merged.